### ISSUE TITLE
dotenv is not working on v1.0.4

### ISSUE THREAD

**Issue by albertilagan** [cachix/devenv#1135]
**Describe the bug**
```
{
   dotenv.enable = true;
}
```
is not working after upgrading from `v1.0.3` to `v1.0.4`

**To reproduce**
1. upgrade `devenv` to `v1.0.4`,
2. make sure `dotenv.enable` is set to `true`,
3. run `devenv up`,
4. envs from `.env` is getting picked up.

**Version**
`devenv 1.0.4 (aarch64-darwin)`

**Comment by DaniruKun** [comment]
Same here

**Comment by sandydoo** [comment]
@albertilagan, I can't seem to replicate this. Am I missing anything?

This the setup I used: https://gist.github.com/sandydoo/640404ce7f88a0f5bbf3c27875ee1453

**Comment by MikeGeorgeff** [comment]
I'm having the same issue, configuring the environment using a flake

**Comment by sandydoo** [comment]
If you're experiencing this issue, please post gists with your `devenv.nix`, `devenv.yaml`, and `devenv.lock` files.

**Comment by prepor** [comment]
@chocoladisco most probably it's because you use flake. And git flakes are not coping flies that are not added to git. Try to exclude .env from .gitignore and `git add .env`. It should "fix" an issue.

The other question is that you most probably don't want to include .env to git. I would love to know proper solution for this

**Comment by sandydoo** [comment]
> most probably it's because you use flake. And git flakes are not coping flies that are not added to git.

Correct. This is well-known design problem with flakes. I don't believe we have any workarounds for this, so if you rely on the dotenv integration, I'd recommend using the devenv cli.

**Comment by prepor** [comment]
Yes, that is what I came up with eventually `nodePackages.dotenv-cli` + `dotenv` in `.envrc` after `use flake ...`

**Comment by sandydoo** [comment]
Closing this because we can't get a response from the OP.

If you're using flakes, the dotenv integration will not work. A Nix flake requires you to track all of the files you need in your config. This is a design limitation/flaw of Nix flakes.

Use devenv's `enterShell` or direnv's `.envrc` to source your .env files.
